applications/CompressiblePotentialFlowApplication/custom_processes/apply_far_field_process.cpp

```diff
@@ -0,0 +1,128 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Inigo Lopez and Marc Nunez
+//
+
+// Project includes
+#include "apply_far_field_process.h"
+
+namespace Kratos {
+
+// Constructor for ApplyFarFieldProcess Process
+ApplyFarFieldProcess::ApplyFarFieldProcess(ModelPart& rModelPart, const double rReferencePotential)
+    : Process(), mrModelPart(rModelPart), mReferencePotential(rReferencePotential)
+{
+    mFreeStreamVelocity = mrModelPart.GetProcessInfo()[FREE_STREAM_VELOCITY];
+}
+
+void ApplyFarFieldProcess::Execute()
+{
+    FindFarthestUpstreamBoundaryNode();
+    AssignFarFieldBoundaryConditions();
+}
+
+void ApplyFarFieldProcess::FindFarthestUpstreamBoundaryNode()
+{
+    // Declaring omp variables, generating vectors of size = num_threads
+    std::size_t num_threads = omp_get_max_threads();
+    std::vector<double> min_projections(num_threads, std::numeric_limits<double>::epsilon());
```

Comment:
I just realized, I believe min_projections should be initialized with a large number.

```suggestion
    std::vector<double> min_projections(num_threads, std::numeric_limits<double>::max());
```

Reply:
ups! Thanks!